### DISCUSSION TITLE
How to cache the last GOP including IDR and non-IDR frames?

### DISCUSSION THREAD

**Discussion by UltraRabbit** [bluenviron/mediamtx#1132] [Questions]
Hi, All,

Is it possible to keep a cache of the latest GOP including IDR and non-IDR frames within the track?  I hope when a reader connected the cached GOP could be sent to the reader before an IDR frame received in real-time stream.  Once the IDR frame got received, the real-time stream could be sent to the reader.   There's a cache of the latest SPS and PPS already, could we also add the cache of a whole latest GOP as well?

I'm not familiar with the source codes yet.  I just locate the codes in streamtrack_h264.go but failed to tell how to intercept the real-time stream forwarding.  Please help me to figure it out.

Thanks in advance,

Jack He

**Comment by aler9** [comment]
Hello, at the moment this is not implemented but it's an interesting feature that would be really useful, i've opened a feature request here #1209

**Comment by aler9** [comment]
PS: please note that this would benefit only unicast users (not UDP-multicast users), but that's better than nothing.
